src/System.Net.Http/src/uap/System/Net/HttpHandlerToFilter.cs

```diff
@@ -47,9 +50,27 @@ internal HttpHandlerToFilter(RTHttpBaseProtocolFilter filter)
             }
             cancel.ThrowIfCancellationRequested();
 
-
             RTHttpRequestMessage rtRequest = await ConvertRequestAsync(request).ConfigureAwait(false);
-            RTHttpResponseMessage rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+
+            RTHttpResponseMessage rtResponse;
+            try
+            {
+                rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+            }
+            catch (TaskCanceledException)
+            {
+                throw;
+            }
+            catch (Exception)
+            {
+                Exception savedEx = (Exception)rtRequest.Properties[SavedExceptionLookupKey];
+                if (savedEx != null)
+                {
+                    throw savedEx;
```

Comment:
Does something higher up the call stack wrap this?  It'd be nice to maintain the original call stack if possible, so if we do need to throw it here, using `ExceptionDispatchInfo.Throw(savedEx)` instead, or capturing the ExceptionDispatchInfo at the original catch site and then throwing that here.